### PR TITLE
feat: adds bitcoind dockerfiles and entrypoint

### PR DESCRIPTION
**Pull request by jurajpiar**
This allows images to be built and pushed to the container registry in order to use them as services inside github actions.

### CODE REVIEW

File: entrypoint.sh [hunk 1]
```diff
@@ -0,0 +1,2 @@
+#!/bin/bash
```
Comment (lucasvuotto) [inline entrypoint.sh:1]: Where is this file used?

File: bitcoind2.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,5 @@
+FROM kylemanna/bitcoind:latest
```
Comment (lucasvuotto) [inline bitcoind2.Dockerfile:1]: The only change between `bitcoind1.Dockerfile` and `bitcoind2.Dockerfile` is the `CMD` line. I suggest you create a single file, `bitcoind.Dockerfile`, with the following content

```
FROM kylemanna/bitcoind:latest
COPY start-bitcoind.sh /usr/local/bin/start-bitcoind.sh
RUN chmod +x /usr/local/bin/start-bitcoind.sh
ENTRYPOINT ["/usr/local/bin/start-bitcoind.sh"]
```

and then it can be run as

```
docker run bitcoind 31591 https://localhost:31592 0.0.0.0:32591 0.0.0.0/0 admin admin
```

for `bitcoind1` and

```
docker run bitcoind 31592 https://localhost:31591 0.0.0.0:32592 0.0.0.0/0 admin admin
```

for `bitcoind2`.

File: config.json [hunk 3]
```diff
@@ -6,7 +6,7 @@
     "pass": "pass"
   },
   "bitcoind": {
-    "url": "bitcoind01",
+    "url": "localhost",
```
Comment (lucasvuotto) [inline config.json]: Is this change intended?

File: entrypoint.sh (hunk 1 above)
Comment (jurajpiar) [reply entrypoint.sh:1]: Oh yea, well spotted, thanks. I renamed it from `start-bitcoin.sh`. Changed now in the dockerfiles.

File: config.json (hunk 3 above)
Comment (jurajpiar) [reply config.json]: yeap, though I'll remove this and do it in situ instad

File: bitcoind2.Dockerfile (hunk 2 above)
Comment (jurajpiar) [reply bitcoind2.Dockerfile:1]: the only reason for these is to be able to build and push in CI https://github.com/jurajpiar/rskj/actions/runs/9059773191/workflow#L149-L170

File: config.json (hunk 3 above)
Comment (jurajpiar) [reply config.json]: change removed

File: bitcoind2.Dockerfile (hunk 2 above)
Comment (jurajpiar) [reply bitcoind2.Dockerfile:1]: unless we can pass the vars to `docker build`? I did not find the way tho.

Comment (lucasvuotto) [reply bitcoind2.Dockerfile:1]: I wasn't aware that GHA doesn't allow to set parameters for service containers. 🙄 

Anyways, you can still do with a single Dockerfile. In fact, you can use directly the `kylemanna/bitcoind` image (although we should migrate at something selfhosted at some point) and then leverage volumes for the service ( https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idservicesservice_idvolumes ) to put the right config file for each container. https://github.com/kylemanna/docker-bitcoind/blob/master/docs/config.md shows where the config file should be mounted to be picked up by `bitcoind` in this container.

Comment (jurajpiar) [reply bitcoind2.Dockerfile:1]: yeap, tried that...

**Comment by jurajpiar** [comment]
I'm going to close this as no longer required by the workflow in rskj.
